frontend/src/components/containers/ContainerHierarchy.js

```diff
@@ -102,31 +127,43 @@ const buildContainerTreeFromPath = (context, path) => {
   if (samples.length) {
     if (!isExploded) {
       children.push({
-        title: <span style={entryStyle}>
+        title: renderEntry(
           <Text type="secondary">
             {samples.length} sample{samples.length === 1 ? '' : 's'}{' '}
             (click to expand)
           </Text>
-        </span>,
+        ),
         icon: <EllipsisOutlined />,
         key: `${container.id}$samples`,
       })
     }
+    else if (isFetching) {
+      children.push({
+        title: renderEntry(
+          <Text type="secondary">
+            {samples.length} sample{samples.length === 1 ? '' : 's'}{' '}
+            (loading)
+          </Text>
+        ),
+        icon: <LoadingOutlined />,
+        key: `${container.id}$samples`,
+      })
+    }
     else {
       children.push(...samples.map(sampleId => {
         const sample = context.samplesByID[sampleId];
-        if (!sample)
+        if (!sample || sample.isFetching)
           return loadingEntry(sampleId);
         const sampleKind = context.sampleKinds.itemsByID[sample.sample_kind]?.name
         return {
-          title: <span style={entryStyle}>
+          title: renderEntry(
             <Link to={`/samples/${sample.id}`}>
               <strong>{sample.name}</strong>{' '}
               sample ({sampleKind})
             </Link>
-          </span>,
+          ),
           icon: <CheckOutlined />,
-          key: sampleId,
+          key: `sample-${sampleId}`,
```

Comment:
This breaks the link to the sample when clicked directly (which uses this key instead of the link defined above). right-click  open in new tab works.

Reply:
As Ulysse said it breaks the link (the link becomes /sample/sample-23 instead of /sample/23), but I dont think the key should be used to generate the url anyways.. The key is a React element that defines a unique HTML DOM element. Could that be fixed without using directly the key?